packages/ui-extensions/src/surfaces/admin/extension-targets.ts

```diff
@@ -40,16 +40,21 @@ type ProductConfigurationComponents = AnyComponentBuilder<
 /**
  * See the [list of available components](/docs/api/admin-extensions/components).
  */
-type AllComponents = AnyComponent;
+type AllComponents = AnyComponentBuilder<
+  Omit<
+  Components,
+    'CustomerSegmentTemplate' | 'InternalCustomerSegmentTemplate'
+  >
+>;
```

Comment:
We don't want our target to be a valid target for any other component except CustomerSegmentTemplate and InternalCustomerSegmentTemplate. I asked about this in #help-app-ui and they agreed with this approach -> https://shopify.slack.com/archives/C04K7BZDH3N/p1694624017224519